lib/ohai/plugins/habitat.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+#
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+Ohai.plugin(:Habitat) do
+  provides "habitat"
+
+  def fetch_habitat_version
+    shell_out("hab -V").stdout.gsub(/hab\s*/, "").strip
+  rescue Ohai::Exceptions::Exec
+    logger.trace("Plugin Habitat: No detected version of hab binary found in PATH, skipping collection.")
+  end
+
+  def fetch_habitat_packages
+    if Dir.exist?("C:/hab/pkgs")
+      Dir.glob("C:/hab/pkgs/*/*/*/*/").sort.map { |pkg| pkg.gsub("C\:\/hab\/pkgs\/", "").chomp("/") }
```

Comment:
Do you have an example of this dir structure that would require us to go this deep down?

Reply:
If you're just trying to get the children in the dirs w/o the full paths you can do Dir.children(FOO). That may allow you to make the hab dirs a per platform helper and condense a large amount of this code down to just a few children method calls.

Reply:
Yeah this structure matches what would be for:
/hab/pkgs/pkg_origin/pkg_name/pkg_version/pkg_release

Trying to avoid shelling out as much as possible and habitat packages will always have these 4 directory paths as part of their tree and is a reliable way to show the installed packages.  Want to capture all 4 of these paths as well because you really need to see everything from origin->release to show the version that is installed for a package.
This will feed back items in to an array that look like `origin/package/version/release`

Reply:
I didnt just `**/*` on it though because only want to capture these 4 directories, nothing below, that is why it is explicitly `/hab/pkgs/*/*/*/*`

Reply:
I could redo this just to do a `hab pkg list --all` and get the output with a rescue though too

Reply:
Yeah ideally we run the command. That would be more reliable with target mode too where we can't just use these ruby file helpers.

Reply:
Cool -- will refactor this a bit, quite a bit actually so i'll close out this PR and create a new one with the updates